### DISCUSSION TITLE
Error Upload to ESP 8266-01

### DISCUSSION THREAD

**Discussion by macoto123** [esp8266/Arduino#7913] [Q&A]
Hey guys,

I have a problem with my Arduino Nano and ESP8266-01S and I tried a few things and nothing worked, I hope someone can help me.. 
Here my problem: I tried to setup an Internet communication with an Arduino Nano and an ESP8266-01. And something is not okay with my circuit I think. 
The pins are connected as follows:
ESP:               Arduino Nano:
VCC ------>   Vin (External Power Supply 3,3V)
RST ------>    none
CHPD ------> Vin (External Power Supply 3,3V)
TXD ------>    TX1
RXD ------>    RX0
GPIO0 ------> GND
GPIO2 ------> none
GND ------> GND

Also I connected the RST from the Nano to GND (Picture follows)

So when I want to Upload an empty Code I get these Errors:
warning: espcomm_sync failed
error: espcomm_open failed
error: espcomm_upload_mem failed
error: espcomm_upload_mem failed

Some background information: 
- I downgraded my esp8266 board drivers to version 2.3.0 (because someone said maybe it would work)
- I tried to connect GPIO0 with a button which is connected to ground and pressed it while plugging in the USB cable
- I tried to lower the baudrate to 57600

Also I measured the Input Voltage on the ESP8266 Board at VCC and Ground and I saw it is always between 2,6 V - 3 V (maybe the wires are crap?)

I am very thankful for every suggestions :)

![2021-03-08 10_58_50-](https://user-images.githubusercontent.com/80256094/110313478-63570480-7ffe-11eb-805e-00c946f59ddc.png)
![IMG_4893](https://user-images.githubusercontent.com/80256094/110313581-8bdefe80-7ffe-11eb-97be-2b756a94856f.jpg)

**Comment by JiriBilek** [comment]
I don't know your knowledge in Arduino world, but the easiest way to use ESP as a wifi communication device is to upload a Espressif AT-command firmware and connect it via USART to your MCU. 
This solution has nothing to do with this repository, though.

**Reply by macoto123** [reply]
yeah unfortunately I don't know how to do this, but is this the right way:
https://www.youtube.com/watch?v=yla44_KZ7eo&ab_channel=ElectronicClinic ?

**Comment by JiriBilek** [comment]
The video shows a kind of a hack where the user is using Arduino Uno's USB-to-Serial converter for connecting the PC to a ESP. You could do the same trick using any USB-to-Serial adapter like e.g. https://www.amazon.com/HiLetgo-FT232RL-Converter-Adapter-Breakout/dp/B00IJXZQ7C. There are tons of such adapters in every shop (Amazon, Aliexpress, your local arduino store, ...)
But I am not sure you actually need to reflash the ESP. All my ESP's came with preloaded AT firmware. If it's your case as well, you'll need only to correctly connect the Arduino to the ESP (most probably you will want to use a SoftwareSerial library). Look here: https://www.instructables.com/Using-ESP-01-and-Arduino-UNO/
Of course, there are libraries that help you to communicate with the ESP and they avoid you to program the complexe AT-command communication. For instance: https://github.com/jandrassy/WiFiEspAT

**Reply by macoto123** [reply]
Thank you very much for your advice, but i cannot get my code as I want to. I just ordered a nodemcu, hopefully it is better for me.
